### PR TITLE
Tests: add tests for EditCommand and EditCommandParser

### LINKED ISSUE #154: Test: add tests for EditCommand

### PR DESCRIPTION
**Pull request by aidanaden**
Closes #154 and #155.

Changes include:
- add `StringUtil.isAlphaNumeric` method to be used in `Name.isValidName` check due to cases where user enters `edit -g 1 -n <new_name>`, edit command assumes `-g 1` to be the **name** of the recipe the user is trying to specify, hence adding a isAlphaNumeric check removes cases where invalid characters/values are entered as a recipe name.
- added toString functions for EditCommand and EditRecipeDescriptor for testing purposes (will remove once tests are finalized)
- updated "Fish & Chips" to "Fish and Chips" in TypicalRecipe and JsonSerializableRecipeBookTest file, also updated NameTest with alpha-numeric names
- added option for empty tags, e.g. `-t ` in scenario where user wants to remove/reset tags from the recipe in edit command

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/DeleteCommand.java [hunk 1]
```diff
@@ -117,7 +117,7 @@ private Recipe getRecipe(List<Recipe> lastShownList, Index recipeIndex) throws C
             return lastShownList.get(zeroBasedIndex);
         }
 
-        throw new CommandException(Messages.MESSAGE_INVALID_RECIPE_DISPLAYED_INDEX);
+        throw new CommandException(Messages.MESSAGE_INVALID_RECIPE_INDEX);
```
Comment (yusufaine) [inline src/main/java/seedu/address/logic/commands/DeleteCommand.java:120]: I'm assuming this change was only to the variable name to make it clearer?

File: src/main/java/seedu/address/logic/commands/EditCommand.java [hunk 2]
```diff
@@ -167,6 +167,15 @@ public boolean equals(Object other) {
...
 
+    @Override
+    public String toString() {
+        return "EditCommand{"
+                + "targetName=" + targetName
+                + ", targetIndex=" + targetIndex
+                + ", editRecipeDescriptor=" + editRecipeDescriptor
+                + '}';
+    }
+
```
Comment (yusufaine) [inline src/main/java/seedu/address/logic/commands/EditCommand.java:178]: Just generally asking if we ever need a toString for any of the commands

File: src/main/java/seedu/address/model/recipe/Name.java [hunk 3]
```diff
@@ -3,13 +3,15 @@
...
+import seedu.address.commons.util.StringUtil;
+
 /**
  * Represents a Recipe's name in the recipe book.
  * Guarantees: immutable; is valid as declared in {@link #isValidName(String)}
  */
 public class Name {
 
-    public static final String MESSAGE_CONSTRAINTS = "Names should not be left blank";
+    public static final String MESSAGE_CONSTRAINTS = "Names should contain valid alpha-numeric characters.";
```
Comment (yusufaine) [inline src/main/java/seedu/address/model/recipe/Name.java:14]: What was the reason why we can accept symbols again?

File: src/test/java/seedu/address/testutil/TypicalRecipes.java [hunk 4]
```diff
@@ -10,64 +10,117 @@
  */
 public class TypicalRecipes {
 
-    public static final Recipe AGLIO_OLIO = new RecipeBuilder().withName("Aglio Olio")
-            .withCompletionTime(5).withIngredients("Garlic 3.0 Cloves", "Spaghetti 1.0", "Olive Oil 500.0 ml",
+    public static final Recipe AGLIO_OLIO = new RecipeBuilder()
```
Comment (yusufaine) [inline src/test/java/seedu/address/testutil/TypicalRecipes.java:13]: Nice catch on the formatting, it makes things easier to read. 👍

**Review by yusufaine — APPROVED** [review]
Overall LGTM, left some comments on some parts I need clarification on.

🌮 🔥 🍝

File: src/main/java/seedu/address/model/recipe/Name.java (hunk 3 above)
Comment (aidanaden) [reply src/main/java/seedu/address/model/recipe/Name.java:14]: No idea actually LOL we only checked for blank names initially, wanted to allow "&" also but one of the test case recipe name was "James&" so decided to just stick to alpha-numeric only

File: src/main/java/seedu/address/logic/commands/DeleteCommand.java (hunk 1 above)
Comment (aidanaden) [reply src/main/java/seedu/address/logic/commands/DeleteCommand.java:120]: yep!

File: src/main/java/seedu/address/logic/commands/EditCommand.java (hunk 2 above)
Comment (aidanaden) [reply src/main/java/seedu/address/logic/commands/EditCommand.java:178]: Only use case is to make testing easier - can print out the values in the EditCommand to check which values didn't get parsed properly etc

Can remove once tests are finalised!

File: src/main/java/seedu/address/model/recipe/Name.java (hunk 3 above)
Comment (yusufaine) [reply src/main/java/seedu/address/model/recipe/Name.java:14]: Should recipe names be allowed to have those though? Shouldn't something like "Grandma's Special Fish & Chips" be valid?

File: src/main/java/seedu/address/logic/commands/EditCommand.java (hunk 2 above)
Comment (yusufaine) [reply src/main/java/seedu/address/logic/commands/EditCommand.java:178]: @aidanaden oh then i think it's fine ah I think

**Review by blaketingyu — APPROVED** [review]
LGTM 🌮

File: src/main/java/seedu/address/model/recipe/Name.java (hunk 3 above)
Comment (aidanaden) [reply src/main/java/seedu/address/model/recipe/Name.java:14]: Potential issues from allowing special characters is names like "grandma recipe'&'" is allowed, which could be smth we could allow??? need discuss i think

**Pull request merged**